hypervisor/include/arch/x86/assign.h

```diff
@@ -9,18 +9,149 @@
+int ptdev_msix_remap(struct acrn_vm *vm, uint16_t virt_bdf, uint16_t entry_nr, struct ptdev_msi_info *info);
+
+
+/**
+ * @brief INTx remapping for passthrough device.
+ *
+ * Set up the remapping of the given virtual pin for the given vm.
+ * Thsi is the main entry for PCI/Legacy device assignment with INTx, calling from vIOAPIC or vPIC.
+ *
+ * @param[in] vm pointer to acrn_vm
+ * @param[in] virt_pin virtual pin number associated with the passthrough device
+ * @param[in] vpin_src ioapic or pic
+ *
+ * @return 0 - on success
+ * @return -ENODEV - for SOS, the entry already be held by others; for UOS, no pre-hold mapping found.
+ *
+ * @pre vm != NULL
+ *
+ */
+int ptdev_intx_pin_remap(struct acrn_vm *vm, uint8_t virt_pin, enum ptdev_vpin_source vpin_src);
+
+/**
+ * @brief Add an interrupt remapping entry for INTx as pre-hold mapping.
+ *
+ * Except vm0, Device Model should call this function to pre-hold ptdev intx
+ * The entry is identified by phys_pin, one entry vs. one phys_pin.
+ * Currently, one phys_pin can only be held by one pin source (vPIC or vIOAPIC).
+ *
+ * @param[in] vm pointer to acrn_vm
+ * @param[in] virt_pin virtual pin number associated with the passthrough device
+ * @param[in] phys_pin physical pin number associated with the passthrough device
+ * @param[in] pic_pin true for pic, false for ioapic
+ *
+ * @return 0 - on success
+ * @return -EINVAL - invalid virt_pin value
+ * @return -ENODEV - failed to add the remapping entry
+ *
+ * @pre vm != NULL
+ *
+ */
+int ptdev_add_intx_remapping(struct acrn_vm *vm, uint8_t virt_pin, uint8_t phys_pin, bool pic_pin);
+
+/**
+ * @brief Remove an interrupt remapping entry for INTx.
+ *
+ * Deactive & remove mapping entry of the given virt_pin for given vm.
+ *
+ * @param[in] vm pointer to acrn_vm
+ * @param[in] virt_pin virtual pin number associated with the passthrough device
+ * @param[in] pic_pin true for pic, false for ioapic
+ *
+ * @return None
+ *
+ * @pre vm != NULL
+ *
+ */
 void ptdev_remove_intx_remapping(const struct acrn_vm *vm, uint8_t virt_pin, bool pic_pin);
-int ptdev_add_msix_remapping(struct acrn_vm *vm, uint16_t virt_bdf,
-	uint16_t phys_bdf, uint32_t vector_count);
-void ptdev_remove_msix_remapping(const struct acrn_vm *vm, uint16_t virt_bdf,
-		uint32_t vector_count);
+
+/**
+ * @brief Add interrupt remapping entry/entries for MSI/MSI-x as pre-hold mapping.
+ *
+ * Add pre-hold mapping of the given number of vectors between the given physical and virtual BDF for the given vm.
+ * Except vm0, Device Model should call this function to pre-hold ptdev MSI/MSI-x.
+ * The entry is identified by phys_bdf:msi_idx, one entry vs. one phys_bdf:msi_idx.
+ *
+ * @param[in] vm pointer to acrn_vm
+ * @param[in] virt_bdf virtual bdf associated with the passthrough device
+ * @param[in] phys_bdf physical bdf associated with the passthrough device
+ * @param[in] vector_count number of vectors
+ *
+ * @return 0 - on success
+ * @return -ENODEV - failed to add the remapping entry
```

Comment:
Same comment as above, please combine the multiple `@return` values into a bullet list